src/System.IO.Abstractions.TestingHelpers/MockFileSystem.cs

```diff
@@ -124,6 +140,28 @@ private string GetPathWithCorrectDirectoryCapitalization(string fullPath)
             return fullPath.TrimSlashes();
         }
 
+        /// <inheritdoc />
+        public MockFileData AdjustTimes(MockFileData fileData, TimeAdjustments timeAdjustments)
+        {
+            var now = (dateTimeProvider ?? defaultDateTimeProvider)();
```

Comment:
Why is this check needed here again? The dateTimeProvider is initialized to the defaultDateTimeProvider, so the below should be fine.
```suggestion
            var now = dateTimeProvider();
```

Reply:
It is not initialized, when serialized/deserialized. I am not sure how to work around this any other way 😟...

Reply:
According to [this thread on StackOverflow](https://stackoverflow.com/a/9419943) I could add the following method to initialize the callback:

```csharp
void OnDeserializing(System.Runtime.Serialization.StreamingContext c);
```

I will try this approach together with the other review changes...

Reply:
Just checked and the approach with the `OnDeserializing` method works. Will change it accordingly...